pkg/operator/ceph/file/controller.go

```diff
@@ -245,15 +245,19 @@ func (r *ReconcileCephFilesystem) reconcile(request reconcile.Request) (reconcil
 
 	// DELETE: the CR was deleted
 	if !cephFilesystem.GetDeletionTimestamp().IsZero() {
-		deps, err := CephFilesystemDependents(r.context, r.clusterInfo, cephFilesystem)
+
+		_, err := cephclient.GetFilesystem(r.context, r.clusterInfo, cephFilesystem.Name)
```

Comment:
I think checking to see if the fs exists is fine. It will prevent issues also where the FS couldn't be properly created and then a user wishes to delete the CR (good info for body text).

In order to merge, the commit message needs fixed to meet spec. It needs a subject that is less than 50 chars long and a body explaining what is being changed and why. The PR title and description should be updated with the new content of the commit once it has been updated. 

The code could be a bit tidier. I would suggest putting the `GetFilesystem()` call inside of `GetFilesystemDependents()` since `GetFilesystem()` is effectively checking whether it's safe to check for dependents. Also keep in mind with `GetFilesystemDependents()` that Rook checks for other CRs, and those can still be checked even when the fs doesn't exist. Once these changes are moved to `GetFilesystemDependents()` I think that will be easier to see and understand the side-effects of.

Thanks for the PR. It's a small corner case, but it helps user experience for cases where it's encountered.

Reply:
@BlaineEXE Thanks for your suggestion, I will update it.